Python/_warnings.c

```diff
@@ -1356,8 +1367,13 @@ PyErr_WarnExplicitFormat(PyObject *category,
         PyObject *res;
         PyThreadState *tstate = get_current_tstate();
         if (tstate != NULL) {
+            WarningsState *st = warnings_get_state(tstate->interp);
```

Comment:
@erlend-aasland - this introduced some warnings in optimized, non-free-threaded builds: unused variable ‘st’

Reply:
I'll get to it.

Reply:
PR: #117373